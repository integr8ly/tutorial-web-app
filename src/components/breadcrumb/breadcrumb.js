import React from 'react';
import PropTypes from 'prop-types';
import { translate } from 'react-i18next';
import { Icon, Breadcrumb as PfBreadcrumb } from 'patternfly-react';
import { withRouter } from 'react-router-dom';

class Breadcrumb extends React.Component {
  homeClicked = () => {
    const { homeClickedCallback } = this.props;
    this.props.history.push('/home');
    if (homeClickedCallback) {
      homeClickedCallback();
    }
  };
  render() {
    const { t, threadName, threadId, totalTasks, taskPosition } = this.props;
    return (
      <PfBreadcrumb className="integr8ly-breadcrumb">
        <PfBreadcrumb.Item onClick={this.homeClicked}>
          <Icon className="fa-lg" type="fa" name="home" />
        </PfBreadcrumb.Item>
        {threadName && !taskPosition && <PfBreadcrumb.Item active>{threadName}</PfBreadcrumb.Item>}
        {threadName &&
          taskPosition && (
            <React.Fragment>
              <PfBreadcrumb.Item href={`/tutorial/${threadId}`}>{threadName}</PfBreadcrumb.Item>
              <PfBreadcrumb.Item active>{t('breadcrumb.task', { taskPosition, totalTasks })}</PfBreadcrumb.Item>
            </React.Fragment>
          )}
      </PfBreadcrumb>
    );
  }
}

Breadcrumb.propTypes = {
  history: PropTypes.object,
  t: PropTypes.func.isRequired,
  threadName: PropTypes.string,
  threadId: PropTypes.string,
  taskPosition: PropTypes.number,
  totalTasks: PropTypes.number,
  homeClickedCallback: PropTypes.func
};

Breadcrumb.defaultProps = {
  history: {},
  threadName: '',
  threadId: null,
  taskPosition: null,
  totalTasks: null,
  homeClickedCallback: undefined
};

const RoutedBreadcrumb = withRouter(translate()(Breadcrumb));

export { RoutedBreadcrumb as default, Breadcrumb };
