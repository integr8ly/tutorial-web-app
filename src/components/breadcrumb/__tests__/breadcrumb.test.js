import { shallow } from 'enzyme';
import React from 'react';
import { Breadcrumb } from '../breadcrumb';

describe('Breadcrumb component', () => {
  const props = {
    history: [],
    t: (localString, localObj) => {
      const { taskPosition, totalTasks } = localObj;
      expect(localString).toEqual('breadcrumb.task');
      return `Task ${taskPosition} of ${totalTasks}`;
    },
    threadName: 'Unit Test Walkthrough',
    threadId: 1,
    taskPosition: 1,
    totalTasks: 3,
    homeClickedCallback: jest.fn()
  };

  test('should render default breadcrumb', () => {
    console.error = jest.fn();
    const component = shallow(<Breadcrumb />);
    expect(component).toMatchSnapshot();
    expect(console.error).toBeCalled();
    console.error.mockRestore();
  });

  test('should render breadcrumb with props', () => {
    const component = shallow(<Breadcrumb {...props} />);
    expect(component).toMatchSnapshot();
  });

  test('should handle home clicked', () => {
    const component = shallow(<Breadcrumb {...props} />);
    component.find('#breadcrumb-home').simulate('click');
    expect(props.homeClickedCallback.mock.calls).toHaveLength(1);
    expect(component).toMatchSnapshot();
  });
});
