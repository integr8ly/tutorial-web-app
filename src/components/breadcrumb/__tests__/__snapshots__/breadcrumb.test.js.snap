// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component should handle home clicked 1`] = `
<Breadcrumb
  aria-label="breadcrumb"
  className=""
>
  <BreadcrumbItem
    className=""
    component="a"
    id="breadcrumb-home"
    isActive={false}
    onClick={[Function]}
    target={null}
    to={null}
  >
    <Icon
      className="fa-lg"
      name="home"
      type="fa"
    />
  </BreadcrumbItem>
  <BreadcrumbItem
    className=""
    component="a"
    isActive={false}
    target={null}
    to="/tutorial/1"
  >
    Unit Test Walkthrough
  </BreadcrumbItem>
  <BreadcrumbItem
    className=""
    component="a"
    isActive={true}
    target={null}
    to={null}
  >
    Task 1 of 3
  </BreadcrumbItem>
</Breadcrumb>
`;

exports[`Breadcrumb component should render breadcrumb with props 1`] = `
<Breadcrumb
  aria-label="breadcrumb"
  className=""
>
  <BreadcrumbItem
    className=""
    component="a"
    id="breadcrumb-home"
    isActive={false}
    onClick={[Function]}
    target={null}
    to={null}
  >
    <Icon
      className="fa-lg"
      name="home"
      type="fa"
    />
  </BreadcrumbItem>
  <BreadcrumbItem
    className=""
    component="a"
    isActive={false}
    target={null}
    to="/tutorial/1"
  >
    Unit Test Walkthrough
  </BreadcrumbItem>
  <BreadcrumbItem
    className=""
    component="a"
    isActive={true}
    target={null}
    to={null}
  >
    Task 1 of 3
  </BreadcrumbItem>
</Breadcrumb>
`;

exports[`Breadcrumb component should render default breadcrumb 1`] = `
<Breadcrumb
  aria-label="breadcrumb"
  className=""
>
  <BreadcrumbItem
    className=""
    component="a"
    id="breadcrumb-home"
    isActive={false}
    onClick={[Function]}
    target={null}
    to={null}
  >
    <Icon
      className="fa-lg"
      name="home"
      type="fa"
    />
  </BreadcrumbItem>
</Breadcrumb>
`;
